service/resource/endpoint/current.go

```diff
@@ -51,5 +51,5 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 			}
 		}
 	}
-	return currentEndpoint, nil
+	return &currentEndpoint, nil
```

Comment:
So this here now dispatches your custom structure but other methods dispatches the k8s endpoint. Why is this different? IMO a resource implementation should have one state it operators on and dispatches through. This is also something we agreed on from the very beginning.

Reply:
This issue evolved from different feedback I got during reviews. `current` and `desired` now dispatch the custom endpoint structure, `New*Patch` receives the custom structure and the private methods `new*Change` do all the checks/changes and then convert it to the k8s endpoints. These now get simply applied by the `Apply*Change` methods. 

The issue is, that the k8s endpoint structure has the IPs which we care about deeply nested. Handling them everywhere got hard to read very quickly, so reviews asked to use a custom structure that only has the interesting data.

Using this custom structure everywhere, forces me to convert the custom structure to the k8s endpoint in the `Apply*Change` methods. We agreed that these methods should _only_ apply though, so the conversion ended up where it is now.

 I would like to fix this and make it consistent. Would a private methods `apply*Change` for conversion be the way to go here? WDYT?

Reply:
Don't know about the details but we should dispatch one single type. I was under the impression it is possible to just compute the k8s endpoint type based on the custom endpoint type. AFAIK there are already convenience functions for this so I would expect this to be quite easy now.